### PR TITLE
imagemagick: new variant ghostscript

### PR DESCRIPTION
**Pull request by wdconinc**
Ghostscript adds about 75 dependencies to an installation of just imagemagick (97 without, 172 with Ghostscript). This adds a variant (defaulting to true for backward compatibility) that allows users to turn off Ghostscript support.

### CODE REVIEW

**Comment by wdconinc** [comment]
> 172 with Ghostscript

Including both mesa and rust...

**Review by tldahlgren — DISMISSED** [review]
LGTM

File: var/spack/repos/builtin/packages/imagemagick/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/imagemagick/package.py:42]: Curious.  Do you also need `--with-gslib` or is that "automatic" with `-with-gs-font-dir`?

Comment (wdconinc) [reply var/spack/repos/builtin/packages/imagemagick/package.py:42]: It defaults to `with`, but you're right that it is better to be explicit.
```suggestion
        if spec.satisfies("+ghostscript"):
            args.append("--with-gslib")
            gs_font_dir = join_path(spec["ghostscript-fonts"].prefix.share, "font")
            args.append("--with-gs-font-dir={0}".format(gs_font_dir))
```

File: var/spack/repos/builtin/packages/imagemagick/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/imagemagick/package.py]:
```suggestion
```
Fix suggestion fail :-(

Comment (tldahlgren) [inline var/spack/repos/builtin/packages/imagemagick/package.py]: BTW. If `font` is a subdirectory, you could just use `spec["ghostscript-fonts"].prefix.share.font`.

**Comment by adamjstewart** [comment]
I would personally default to false to reduce build times...

**Comment by wdconinc** [comment]
I agree, but we would need to get involvement from all dependent maintainers since they may be implicitly assuming that imagemagick always has ghostscript support. 

Let me tag them when I'm at a PC and can run that on a command line.

**Comment by adamjstewart** [comment]
According to:
```console
$ spack maintainers $(spack dependents imagemagick)
```
it looks like  @glennpj @mtmiller  @pradyunsg  @siko1056 may have opinions since they maintain packages that depend on imagemagick.

Specific packages are:
```
meme  octave  py-matplotlib  py-wand  r-animation  r-magick  r-webshot
```

File: var/spack/repos/builtin/packages/imagemagick/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/imagemagick/package.py]: FYI. `join_path` is no longer needed in this case.

```suggestion
            gs_font_dir = spec["ghostscript-fonts"].prefix.share.font
```

**Comment by adamjstewart** [comment]
Alternatively, if you don't want to wait for the above people to review this PR, you can try building those 7 packages and make sure they work with `imagemagick~ghostscript`.

**Comment by glennpj** [comment]
I am okay with this change.

**Comment by wdconinc** [comment]
I verified that all dependents build successfully without the ghostscript functionality.

I suspect the main impact will be that these dependent packages will not be able to save figures as `eps` (and maybe `pdf`) anymore unless they add `+ghostscript`. I tested with octave and it still writes to `pdf`, but not `eps`:
```
figure (1);
clf ();
surf (peaks);
print figure1.pdf
print figure1.eps
```

**Review by tldahlgren — DISMISSED** [review]

**Comment by cessenat** [comment]
> 

Concerning octave, which is of major interest for me, the default is that imagemagick support is not built.
To me, your PR is fair, and @adamjstewart suggestion to not enable ghostscript by default is fair.

**Comment by pradyunsg** [comment]
I don't have any opinions on this. :)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
